test/runnable/test19731.d

```diff
@@ -0,0 +1,53 @@
+// PERMUTE_ARGS:
+struct Foo
+{
+    Object obj_;
+
+    invariant (obj_ !is null);
+
+    auto obj7()
+    {
+        return this.obj_;
```

Comment:
For LDC, the `this` VarDeclaration's parent2 isn't the codegen'd `Foo.obj7` anymore (so we wrongly assume it's a nested variable). It's another `obj7`, so I guess the one from the first semantic pass...